File: other/EducationalDPContest/K/K.py
N,K=map(int,input().split())
A=list(map(int,input().split()))
dp=[0]*(K+1)
for i in range(A[0]):
    dp[i]=1
for i in range(K+1):
    for j in range(N):
        if i+A[j]>K:
            break
        if dp[i+A[j]]>=0:
            dp[i+A[j]]=dp[i]*(-1)
if dp[K]==1:
    print("Second")
elif dp[K]==-1:
    print("First")        

File: other/EducationalDPContest/K/K.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int main()
{
	ll N, K;
	cin >> N >> K;
	vector<ll> A(N);
	for (ll i = 0; i < N; i++)
		cin >> A[i];
	vector<bool> dp(K + 1, false);
	for (ll i = 0; i <= K; i++)
	{
		if (dp[i])
			continue;
		for (ll j = 0; j < N; j++)
		{
			if (i + A[j] <= K)
				dp[i + A[j]] = true;
		}
	}
	if (dp[K])
		cout << "First" << endl;
	else
		cout << "Second" << endl;
}
